lib/plugins/aws/package/compile/events/s3/index.js

```diff
@@ -10,32 +10,34 @@ class AwsCompileS3Events {
     this.options = options;
     this.provider = this.serverless.getProvider('aws');
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8031
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 's3', {
-      anyOf: [{ type: 'string' }, { type: 'object' }],
+      anyOf: [
+        { $ref: '#/definitions/awsS3BucketName' },
```

Comment:
Here we also should support just `string` as it also can be reference to bucket data in `provider.s3` group

Reply:
`awsS3BucketName` is a string, so I think adding `string` to the `anyOf` array will make `awsS3BucketName` redundant. I will replace.

Reply:
Yes, I was thinking about replacing it with plain `string` requirement. Sorry I wasn't probably clear enough